spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraReader.scala

```diff
@@ -0,0 +1,166 @@
+package com.datastax.spark.connector.rdd
+
+import java.io.IOException
+
+import com.datastax.driver.core.{ProtocolVersion, Session}
+import com.datastax.spark.connector._
+import com.datastax.spark.connector.cql.{CassandraConnector, Schema}
+import com.datastax.spark.connector.rdd.reader._
+
+import scala.reflect.ClassTag
+
+trait CassandraReader[R] {
+
+
+  protected def connector: CassandraConnector
```

Comment:
I guess many of those protected methods here are just declared in order to get access to their implementations in CassandraRDD. Additionally, this trait is used as the only building block for CassandraRDD subclasses, it doesn't make sense to use it outside of CassandraRDD hierarchy. Therefore, you can use this nice Scala feature:
```scala    
   trait CassandraReader[R] {
      this: CassandraRDD[_] => 
```